### PR TITLE
Handle section concordance

### PR DESCRIPTION
**Pull request by scott-ace-newton**

### CODE REVIEW

File: concept/service.go [hunk 1]
```diff
@@ -70,80 +81,93 @@ func (s *AggregateService) ListenForNotifications() {
...
+		log.WithFields(log.Fields{"UUID": concordedConcept.PrefUUID, "transaction_id": transactionID}).Debug("Sending update notification to kinesis of update to concept: " + updatedId)
+	}
+
+	log.WithFields(log.Fields{"UUID": concordedConcept.PrefUUID, "transaction_id": transactionID}).Info("Finished processing update")
 
 	return nil
 }
 
-func (s *AggregateService) GetConcordedConcept(UUID string) (ConcordedConcept, string, error) {
+func (s *AggregateService) GetConcordedConcept(UUID string) (ConcordedConcept, string, error, string, httpStatus) {
```
Comment (nwrigh) [inline concept/service.go]: Can we rename this errorStatus as this service should be agnostic of transport mechanism

File: concept/service.go [hunk 2]
```diff
@@ -70,80 +81,93 @@ func (s *AggregateService) ListenForNotifications() {
 }
 
 func (s *AggregateService) ProcessMessage(UUID string) error {
-
 	// Get the concorded concept
-	concordedConcept, transactionID, err := s.GetConcordedConcept(UUID)
+	concordedConcept, transactionID, err, _, _ := s.GetConcordedConcept(UUID)
```
Comment (nwrigh) [inline concept/service.go]: Why have the logMsg and http status (which needs renaming) if we are ignoring them?

File: concept/service.go [hunk 3]
```diff
@@ -70,80 +81,93 @@ func (s *AggregateService) ListenForNotifications() {
...
 	if err != nil {
```
Comment (nwrigh) [inline concept/service.go]: How does the handler know to diffientate the err for different status codes? Or do both DOWNSTREAM_ERROR and NOT_FOUND result in the same code in the handler? If so we can get rid of the status altogether

File: concept/service_test.go [hunk 4]
```diff
@@ -110,59 +128,71 @@ type mockHTTPClient struct {
...
 	assert.NoError(t, err)
 	assert.Equal(t, "tid_123", tid)
 	assert.Equal(t, "Test Concept", c.PrefLabel)
 }
 
 func TestAggregateService_GetConcordedConcept_TMEConcordance(t *testing.T) {
-	svc, _, _, _ := setupTestService(200)
+	svc, _, _, _, _ := setupTestService(200)
 
 	expectedConcept := ConcordedConcept{
```
Comment (nwrigh) [inline concept/service_test.go]: We should extend this test case with all the new proerties

File: concept/service_test.go [hunk 5]
```diff
@@ -218,7 +257,7 @@ func TestResolveConceptType(t *testing.T) {
 	assert.Equal(t, "topics", topic)
 }
 
-func setupTestService(httpError int) (Service, *mockS3Client, *mockSQSClient, *mockDynamoDBClient) {
+func setupTestService(httpError int) (Service, *mockS3Client, *mockSQSClient, *mockDynamoDBClient, *mockKinesisStreamClient) {
```
Comment (nwrigh) [inline concept/service_test.go]: We ignore *mockS3Client, *mockSQSClient, *mockDynamoDBClient, return values so really are they needed?

File: concept/service.go (hunk 2 above)
Comment (scott-ace-newton) [reply concept/service.go]: They are used by the GET endpoint as described below

File: concept/service.go (hunk 3 above)
Comment (scott-ace-newton) [reply concept/service.go]: The function is returning the status enum(not http_status, ill rename it) and I have a switch/case statement like our other handlers have. A Downstream error returns a 500 and Not Found returns a 404 which are both valid cases. Rob was seeing 404 no matter what error was thrown hence the need to disambiguate.

File: concept/service_test.go (hunk 5 above)
Comment (scott-ace-newton) [reply concept/service_test.go]: we use mockSQS and mockKinesis in the tests but not the other two. Ill take another look to see if they are needed

**Pull request merged**